components/ai_chat/content/browser/page_content_fetcher.cc

```diff
@@ -267,7 +317,31 @@ class PageContentFetcher {
                             invalidation_token, true);
   }
 
+  void OnPatchFetchResponse(FetchPageContentCallback callback,
+                            std::unique_ptr<network::SimpleURLLoader> loader,
+                            std::unique_ptr<std::string> response_body) {
+    auto response_code = -1;
+    base::flat_map<std::string, std::string> headers;
+    if (loader->ResponseInfo()) {
+      auto headers_list = loader->ResponseInfo()->headers;
+      if (headers_list) {
+        response_code = headers_list->response_code();
+      }
+    }
+    bool is_ok =
+        loader->NetError() == net::OK && response_body && response_code == 200;
+    ;
+    std::string patch_content = is_ok ? *response_body : "";
+    if (!is_ok || patch_content.empty()) {
+      DVLOG(1) << __func__ << " invalid patch response from url: "
+               << loader->GetFinalURL().spec() << " status: " << response_code;
+    }
+    DVLOG(2) << "Got patch: " << patch_content;
+    SendResultAndDeleteSelf(std::move(callback), patch_content, "", false);
```

Comment:
By not passing an invalidation token, I think we're performing these extra fetches unnecessarily every time the user submits a new AI Chat conversation entry. Perhaps we're ok with that because of network caching? If not, and we don't expect the content of the patch url to change during a conversation (i.e. during a Tab's page lifecycle), then we can use the patch url as the invalidation token. Then, if the next call to FetchContent provides the same patch url as is generated from `GetGithubPatchURLForPRURL`, the callback can be called immediately passing the same invalidation token and the caller (ConversationDriver) will know to use its content cache instead.
This is what's done for YouTube to avoid any unnecessary re-fetch and parsing during a conversation, and the plumbing is all setup.

Reply:
Thanks I'll fix that in this follow up https://github.com/brave/brave-browser/issues/36442